packages/app/src/server/service/g2g-transfer.ts

```diff
@@ -231,54 +232,63 @@ export class G2GTransferPusherService implements Pusher {
+  public async getTransferability(destGROWIInfo: IDataGROWIInfo): Promise<Transferability> {
+    const { fileUploadService, configManager } = this.crowi;
 
     const version = this.crowi.version;
-    if (version !== toGROWIInfo.version) {
+    if (version !== destGROWIInfo.version) {
       return {
         canTransfer: false,
         // TODO: i18n for reason
-        reason: `Growi versions mismatch. This Growi: ${version} / new Growi: ${toGROWIInfo.version}.`,
+        reason: `GROWI versions mismatch. src GROWI: ${version} / dest GROWI: ${destGROWIInfo.version}.`,
       };
     }
 
     const activeUserCount = await this.crowi.model('User').countActiveUsers();
-    if ((toGROWIInfo.userUpperLimit ?? Infinity) < activeUserCount) {
+    if ((destGROWIInfo.userUpperLimit ?? Infinity) < activeUserCount) {
+      return {
+        canTransfer: false,
+        // TODO: i18n for reason
+        reason: `The number of active users (${activeUserCount} users) exceeds the limit of dest GROWI (to up ${destGROWIInfo.userUpperLimit} users).`,
+      };
+    }
+
+    if (destGROWIInfo.fileUploadDisabled) {
       return {
         canTransfer: false,
         // TODO: i18n for reason
-        reason: `The number of active users (${activeUserCount} users) exceeds the limit of new Growi (to up ${toGROWIInfo.userUpperLimit} users).`,
+        reason: 'File upload is disabled in dest GROWI.',
       };
     }
 
-    if (toGROWIInfo.fileUploadDisabled) {
+    if (configManager.getConfig('crowi', 'app:fileUploadType') === 'none') {
       return {
         canTransfer: false,
         // TODO: i18n for reason
-        reason: 'File upload is disabled in new Growi.',
+        reason: 'File upload is not configured for src GROWI.',
       };
     }
 
-    if (!toGROWIInfo.attachmentInfo.writable) {
+    const shouldUseFileUploadTypeOfDestGROWI = destGROWIInfo.attachmentInfo.type !== 'none';
+    if (shouldUseFileUploadTypeOfDestGROWI && !destGROWIInfo.attachmentInfo.writable) {
       return {
         canTransfer: false,
         // TODO: i18n for reason
-        reason: 'The storage of new Growi is not writable.',
+        reason: 'The storage of dest GROWI is not writable.',
       };
     }
 
     const totalFileSize = await fileUploadService.getTotalFileSize();
-    if ((toGROWIInfo.fileUploadTotalLimit ?? Infinity) < totalFileSize) {
+    if ((destGROWIInfo.fileUploadTotalLimit ?? Infinity) < totalFileSize) {
       return {
         canTransfer: false,
         // TODO: i18n for reason
         // eslint-disable-next-line max-len
-        reason: `Total file size exceeds file upload limit of new Growi. Requires ${totalFileSize.toLocaleString()} bytes, but got ${(toGROWIInfo.fileUploadTotalLimit ?? Infinity).toLocaleString()} bytes.`,
+        reason: `Total file size exceeds file upload limit of dest GROWI. Requires ${totalFileSize.toLocaleString()} bytes, but got ${(destGROWIInfo.fileUploadTotalLimit ?? Infinity).toLocaleString()} bytes.`,
```

Comment:
```suggestion
        reason: `The total file size of attachments exceeds the file upload limit of the destination GROWI. Requires ${totalFileSize.toLocaleString()} bytes, but got ${destGROWIInfo.fileUploadTotalLimit.toLocaleString()} bytes.`,
```
この if 分に入るときに、Inifinity のケースなくね？

Reply:
defaultが`null`, `Infinity`なので`null`になることがあります。
https://github.com/weseek/growi/blob/f61b9e77e9c9374a2c30b89c780b047005892bd7/packages/app/src/server/service/config-loader.ts#L256-L261
https://github.com/weseek/growi/blob/f61b9e77e9c9374a2c30b89c780b047005892bd7/packages/app/src/server/service/config-loader.ts#L310-L318

型にも`null`が入っています。
https://github.com/weseek/growi/blob/f61b9e77e9c9374a2c30b89c780b047005892bd7/packages/app/src/server/service/g2g-transfer.ts#L59-L60

APIを経由すると`Infinity`が`null`に変わってしまうので`Infinity`に戻しています。
```ts
JSON.stringify(Infinity)
// 'null'
```

Reply:
うーん、if 文で以下のように書いているので、

```
if ((destGROWIInfo.fileUploadTotalLimit ?? Infinity) < totalFileSize) {
```

`fileUploadTotalLimit == null` のときは Infinity になって、この if 文の中には入らないような気がするんだが…？
メッセージに `but got Infinity bytes` って出るのおかしいよね？

```
reason: `The total file size of attachments exceeds the file upload limit of the destination GROWI. Requires ${totalFileSize.toLocaleString()} bytes, but got ${(destGROWIInfo.fileUploadTotalLimit ?? Infinity).toLocaleString()} bytes.`,
```

Reply:
完全理解。`as number`しておきました。